src/services/api.ts

```diff
@@ -0,0 +1,47 @@
+export class APIService {
+  static baseURL: string = process.env.API_URL || 'http://localhost:3000';
+
+  static async signIn(username: string, password: string) {
+    const response = await fetch(`${APIService.baseURL}/auth/login`, {
+      method: 'POST',
+      body: JSON.stringify({ 
+        "username": username, 
+        "password": password 
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      credentials: 'include',
+    });
+    console.log(response);
```

Comment:
Isn't this a debug statement? Should we really be pushing these into `main` branch?

Maybe we can use these kinds of statements with some flag checks that check if the `DEBUG_SIGNIN` mode is on? If we want to use stuff like this we can create a common service for it.

Reply:
I tend to keep the logs just to be sure everything is working. It gives the idea like

If it's 404, the NestJS server may not be working
If the error code is 500, perhaps the SQL server is not running
If it's 401, unauthorized...

Later on before production we can remove all of them. But if you think they are unnecessary, we can directly remove them now. What do you think? @HuseyinSimsek7904

Reply:
I think it is okay to keep them, however, let's at least add an explanation on what this is.

Maybe something like:
```suggestion
    console.log("Received response from signIn API: ", response);
```

Reply:
Cool :+1: